front/src/applications/operationalStudies/hooks/useLazyLoadTrains.ts

```diff
@@ -66,51 +58,44 @@ const useLazyLoadTrains = ({
     setTrainIdsToProject,
   });
 
-  // gradually fetch the simulation of the trains
-  useEffect(() => {
-    const getTrainScheduleSummaries = async (_infraId: number, _trainToFetchIds: number[]) => {
-      setAllTrainsLoaded(false);
-
-      for (let i = 0; i < _trainToFetchIds.length; i += BATCH_SIZE) {
-        const packageToFetch = getBatchPackage(i, _trainToFetchIds, BATCH_SIZE);
-
-        const rawSummaries = await postV2TrainScheduleSimulationSummary({
-          body: {
-            infra_id: _infraId,
-            ids: packageToFetch,
-            electrical_profile_set_id: electricalProfileSetId,
-          },
-        }).unwrap();
-
-        // launch the projection of the trains
-        setTrainIdsToProject((prev) => [...prev, ...packageToFetch]);
-
-        // format the summaries to display them in the timetable
-        const newFormattedSummaries = formatTrainScheduleSummaries(
-          packageToFetch,
-          rawSummaries,
-          trainSchedulesById,
-          rollingStocks
-        );
-
-        // as formattedSummaries is a dictionary, we replace the previous values with the new ones
-        setTrainScheduleSummariesById((prev) => concatMap(prev, newFormattedSummaries));
-      }
-
-      setTrainIdsToFetch([]);
-      setAllTrainsLoaded(true);
-    };
-
-    if (infraId && trainIdsToFetch && trainIdsToFetch.length > 0) {
-      getTrainScheduleSummaries(infraId, trainIdsToFetch);
+  const loadTrainIds = async (trainIds: number[]) => {
+    setAllTrainsLoaded(false);
+
```

Comment:
if we have no scenario we don’t get an infraId from `useScenarioData`. 
in that case we need not do anything and just return early.

```suggestion
  const loadTrainIds = async (trainIds: number[]) => {
    if (!infraId) {
      return;
    }

Reply:
Is this something which can really happen? Can we have a list of trains to load from a timetable without an infra?

If it can't happen, I'd rather fail hard if we hit this case, so that we can debug the root cause instead of silently doing nothing.

Reply:
alright, I think I’m okay to consider we should have a studyId defined at this point